### ISSUE TITLE
make dist doesn't work

### ISSUE THREAD

**Issue by ngena-bl** [ntop/nDPI#738]
I tried to do `make dist` to create a tarball but the process fails.

I'm getting the following error:
```
/home/user/nDPI> make dist
make  dist-gzip am__post_remove_distdir='@:'
make[1]: Entering directory '/home/user/nDPI'
if test -d "libndpi-2.9.0"; then find "libndpi-2.9.0" -type d ! -perm -200 -exec chmod u+w {} ';' && rm -rf "libndpi-2.9.0" || { sleep 5 && rm -rf "libndpi-2.9.0"; }; else :; fi
test -d "libndpi-2.9.0" || mkdir "libndpi-2.9.0"
 (cd src/lib && make  top_distdir=../../libndpi-2.9.0 distdir=../../libndpi-2.9.0/src/lib \
     am__remove_distdir=: am__skip_length_check=: am__skip_mode_fix=: distdir)
make[2]: Entering directory '/home/user/nDPI/src/lib'
make[2]: *** No rule to make target 'distdir'.  Stop.
make[2]: Leaving directory '/home/user/nDPI/src/lib'
Makefile:585: recipe for target 'distdir' failed
make[1]: *** [distdir] Error 1
make[1]: Leaving directory '/home/user/nDPI'
Makefile:684: recipe for target 'dist' failed
make: *** [dist] Error 2
```
Prior running the `make dist` I did the following steps:
I cloned the repository (`git clone https://github.com/ntop/nDPI.git`) (obviously)
I did `./autogen.sh`
I did `./configure`
I did `make`

All these steps were successful.

Using the following environment:
```
/home/user/nDPI/tests> uname -a
Linux user-vm 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by nickbroon** [comment]
My best guess is that this symptomatic of the fact that nDPI/src/lib is not using automake, but appears to a custom Makefile.in committed.
I'm not sure why this is the case, especially as the Makefile.in is populated with values from configure which requires automake anyway.
I think the build system for nDPI needs tidying up and use Makefile.am in all the subdirs. (I note that some subdir already make use of Makefile.am/automake)

**Comment by utoni** [comment]
`make dist` was fixed with 656323c33450db868cda7b3e2cc75d0e417d8d6e
